application/src/main/java/org/togetherjava/tjbot/commands/moderation/AuditCommand.java

```diff
@@ -139,26 +138,58 @@ private boolean handleChecks(@NotNull Member bot, @NotNull Member author,
         return ModerationUtils.handleHasAuthorRole(ACTION_VERB, hasRequiredRole, author, event);
     }
 
-    private void auditUser(@NotNull User user, @NotNull ISnowflake guild,
-            @NotNull IReplyCallback event) {
-        List<ActionRecord> actions =
-                actionsStore.getActionsByTargetAscending(guild.getIdLong(), user.getIdLong());
+    private Message auditMessage(long guildId, long targetId, int pageNo, @NotNull JDA jda) {
+        List<ActionRecord> actions = actionsStore.getActionsByTargetAscending(guildId, targetId);
 
-        MessageEmbed summary = createSummaryMessage(user, actions);
-        if (actions.isEmpty()) {
-            event.replyEmbeds(summary).queue();
-            return;
+        List<List<ActionRecord>> groupedActions = new ArrayList<>();
+        for (int i = 0; i < actions.size(); i++) {
+            if (i % 25 == 0) {
+                groupedActions.add(new ArrayList<>(25));
+            }
+
+            groupedActions.get(groupedActions.size() - 1).add(actions.get(i));
+        }
+
+        int totalPage = groupedActions.size();
+
+        EmbedBuilder audit = createSummaryEmbed(jda.retrieveUserById(targetId).complete(), actions)
+            .setFooter("Page: " + pageNo + "/" + totalPage);
+        groupedActions.get(pageNo - 1)
+            .forEach(action -> audit.addField(actionToField(action, jda)));
+
+        return new MessageBuilder(audit.build())
+            .setActionRows(ActionRow.of(previousButton(guildId, targetId, pageNo),
+                    nextButton(guildId, targetId, pageNo, totalPage)))
+            .build();
+    }
+
+    private Button previousButton(long guildId, long targetId, int pageNo) {
+        Button previousButton = Button.primary(generateComponentId(String.valueOf(guildId),
+                String.valueOf(targetId), String.valueOf(pageNo), "-1"), "⬅");
```

Comment:
What is the -1 and the 1 in nextButton for? If this is to remember which button is which, use an enum instead.

Reply:
this for how much this button increments the page number by

Reply:
Ah. Thats cool. Consider putting it into a well named variable instead.
```java
int turnPageBy = -1;
```
or something like that.

Reply:
i made a new method for this

Reply:
this problem is still there though. To a reader of the code, it is absolutely unclear what 1 and -1 means. You first need to scroll around to the onButtonClick to understand what it does.

Reply:
```java
        String previousButtonTurnPageBy = "-1";
        Button previousButton = Button.primary(generateComponentId(stringGuildId, stringTargetId,
                stringCallerId, stringPageNumber, previousButtonTurnPageBy), "⬅");

        String nextButtonTurnPageBy = "1";
        Button nextButton = Button.primary(generateComponentId(stringGuildId, stringTargetId,
                stringCallerId, stringPageNumber, nextButtonTurnPageBy), "➡");
```

is this better?

Reply:
yes